experimental/iterators/test/Conversion/IteratorsToStandard/iterators-to-standard.mlir

```diff
@@ -0,0 +1,24 @@
+// RUN: mlir-proto-opt %s -convert-iterators-to-std \
+// RUN: | FileCheck %s
+
+func @main() {
+  %input = "iterators.sampleInput"() : () -> (!iterators.iterator<tuple<i32>>)
+  %reduce = "iterators.reduce"(%input) : (!iterators.iterator<tuple<i32>>) -> (!iterators.iterator<tuple<i32>>)
+  "iterators.sink"(%reduce) : (!iterators.iterator<tuple<i32>>) -> ()
+  return
+}
+// CHECK:      module {
+// CHECK-NEXT:   func private @iteratorsMakeSampleInputOperator() -> !llvm.ptr<i8>
+// CHECK-NEXT:   func private @iteratorsDestroySampleInputOperator(!llvm.ptr<i8>)
+// CHECK-NEXT:   func private @iteratorsMakeReduceOperator(!llvm.ptr<i8>) -> !llvm.ptr<i8>
+// CHECK-NEXT:   func private @iteratorsDestroyReduceOperator(!llvm.ptr<i8>)
+// CHECK-NEXT:   func private @iteratorsComsumeAndPrint(!llvm.ptr<i8>)
+// CHECK-NEXT:   func @main() {
+// CHECK-NEXT:     %0 = call @iteratorsMakeSampleInputOperator() : () -> !llvm.ptr<i8>
```

Comment:
Please don't rely on SSA value names as this often breaks.

Favor FileCheck captures if you need to track producer/consumers:
```
%[[V0:.*]] = call @iteratorsMakeSampleInputOperator()
call @iteratorsMakeReduceOperator(%[[V0]])
etc...
```

Reply:
Ah, good pointer! I was wondering how to make this more robust...